### PR TITLE
Password reset

### PR DESCRIPTION
**Pull request by haquangframgia**

### CODE REVIEW

**Comment by haquangframgia** [comment]
Ready

File: app/controllers/password_resets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  def check_valid
+    @user = User.find_by email: params[:email]
+    redirect_to :root unless @user && @user.activated && @user.authenticated?(:reset, params[:id])
+    return unless @user.password_reset_expired?
+    flash[:danger] = t "reset_password.expired_password"
+    redirect_to new_password_reset_url
+  end
+end
```
Comment (huuphuc93) [inline app/controllers/password_resets_controller.rb]: end of file

File: app/controllers/password_resets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  private
+
+  def user_params
+    params.require(:user).permit :password, :password_confirmation
+  end
+
+  def check_valid
+    @user = User.find_by email: params[:email]
+    redirect_to :root unless @user && @user.activated && @user.authenticated?(:reset, params[:id])
```
Comment (huuphuc93) [inline app/controllers/password_resets_controller.rb]: @user&.activated &.authenticated? :reset, params[:id]

File: app/views/user_mailer/password_reset.html.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<%= t("mail.reset_mail_content") %
+        edit_password_reset_url(@user.reset_token, email: @user.email) %>
```
Comment (huuphuc93) [inline app/views/user_mailer/password_reset.html.erb]: cách 2 spaces thoai, % làm gì nhỉ?

**Comment by huuphuc93** [comment]
commented

**Comment by haquangframgia** [comment]
Ready

**Comment by huuphuc93** [comment]
merged

**Pull request merged**